qiskit/optimization/algorithms/optimization_algorithm.py

```diff
@@ -113,13 +113,13 @@ def __init__(self, x: Optional[Any] = None, fval: Optional[Any] = None,
                  results: Optional[Any] = None,
                  status: OptimizationResultStatus = OptimizationResultStatus.SUCCESS,
                  variables: Optional[List[Variable]] = None) -> None:
-        self._x = x if x is not None else []   # pylint: disable=invalid-name
-        self._variables = variables if variables is not None else []
+        self._x = x if x else []   # pylint: disable=invalid-name
```

Comment:
I did not see the build failure reason - was it related to this? Either way the 'is not None' seems like the right solution otherwise valid values can be treated as False which will lead them to becoming the empty list which seems wrong to me.

Reply:
OK, I will revert that to check 'is not None' explicitely,
thanks